hts-precompile/IHederaTokenService.sol

```diff
@@ -441,6 +565,186 @@ interface IHederaTokenService {
+    function getNonFungibleTokenInfo(address token, uint32 serialNumber)
+        external
+        returns (int responseCode, NonFungibleTokenInfo memory nonFungibleTokenInfo);
+
+    /// Operation to freeze token account
+    /// @param token The token address
+    /// @param account The account address to be frozen
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function freezeToken(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to unfreeze token account
+    /// @param token The token address
+    /// @param account The account address to be unfrozen
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function unFreezeToken(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to grant kyc to token account
+    /// @param token The token address
+    /// @param account The account address to grant kyc
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function grantTokenKyc(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to revoke kyc to token account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function revokeTokenKyc(address token, address account)
+        external
+        returns (int responseCode);
+
+    /// Operation to pause token
+    /// @param token The token address to be paused
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function pauseToken(address token) external returns (int responseCode);
+
+    /// Operation to unpause token
+    /// @param token The token address to be unpaused
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function unPauseToken(address token) external returns (int responseCode);
+
+    /// Operation to wipe fungible tokens from account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @param amount The number of tokens to wipe
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function wipeTokenAccount(
+        address token,
+        address account,
+        uint32 amount
+    ) external returns (int responseCode);
+
+    /// Operation to wipe non fungible tokens from account
+    /// @param token The token address
+    /// @param account The account address to revoke kyc
+    /// @param  serialNumbers The serial numbers of token to wipe
+    /// @return responseCode The response code for the status of the request. SUCCESS is 22.
+    function wipeTokenAccount(
```

Comment:
We have to change the name here to be unique. Otherwise we would get the following TypeError: Member "wipeTokenAccount" not unique after argument-dependent lookup in type(contract IHederaTokenService) when we try to make abi.encodeWithSelector(IHederaTokenService.wipeTokenAccount.selector, token, account, amount)); in HederaTokenService.sol

Reply:
@shemnon does this highlight a services bug?
We should be able to support overloading of functions in precompile contracts

Reply:
Backup name is `wipeTokenAccountNFT` if we have to change it

Reply:
This may be a solidity issue.  At the EVM level it is not an issue.

@tdermendzhievv what solidity version did you observe this with?

If we need to change names I would rename the NFT one `wipeTokenAccountNFTs`

Reply:
@shemnon from [WipeTokenAccount.sol](https://github.com/hashgraph/hedera-services/pull/3658/files#diff-c15380743e33fc03880913485860a775e242ddc69460569d533e946677909f23) looks like `0.6.12`
Currently our precompile sol file nots `solidity >=0.4.9 <0.9.0`

Reply:
EVM is fine, but I think we will need to rename and not use overloading.  I think there are some interactions with solidity's `selector` semantics that makes overloading odeus to deal with.

Reply:
Updated to `wipeTokenAccountNFT` to avoid overload concern